### PR TITLE
Fix crash on API <24 and make sure this error will not occur again.

### PR DESCRIPTION
**Pull request by bmarty**
Fix crash on API <24 and make sure this error will not occur again.

### CODE REVIEW

File: library/core-utils/src/main/java/im/vector/lib/core/utils/compat/MutableCollectionCompat.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+package im.vector.lib.core.utils.compat
+
+import android.os.Build
+
+fun <E> MutableCollection<E>.removeIfCompat(predicate: (E) -> Boolean) {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        removeIf(predicate)
+    } else {
+        removeAll(filter(predicate).toSet())
```
Comment (bmarty) [inline library/core-utils/src/main/java/im/vector/lib/core/utils/compat/MutableCollectionCompat.kt:25]: I have been asked to convert to Set for performance reason.

**Review by ouchadam — APPROVED** [review]

File: library/core-utils/src/main/java/im/vector/lib/core/utils/compat/MutableCollectionCompat.kt (hunk 1 above)
Comment (ouchadam) [reply library/core-utils/src/main/java/im/vector/lib/core/utils/compat/MutableCollectionCompat.kt:25]: as this is only for the _keys to remove_ there shouldn't be any changes for lists with duplicated items?

```kotlin
listOf(1, 1, 2).removeIfCompat { it == 2 }  // would still expect [1, 1] 
```

File: vector/lint.xml [hunk 2]
```diff
@@ -40,6 +40,7 @@
     <issue id="RtlSymmetry" severity="error" />
 
     <!-- Code -->
+    <issue id="NewApi" severity="error" />
```
Comment (ouchadam) [inline vector/lint.xml:43]: :100:

File: library/core-utils/src/main/java/im/vector/lib/core/utils/compat/MutableCollectionCompat.kt (hunk 1 above)
Comment (bmarty) [reply library/core-utils/src/main/java/im/vector/lib/core/utils/compat/MutableCollectionCompat.kt:25]: Now that you say it, I am not sure what will happen with
`listOf(1, 1, 2).removeIfCompat { it == 1 }  // would still expect [2] but I think we will get [1, 2]`
Let me play a bit with it.

Comment (bmarty) [reply library/core-utils/src/main/java/im/vector/lib/core/utils/compat/MutableCollectionCompat.kt:25]: Seems to work as expected:
image
(https://pl.kotl.in/vIVLDMXe-)

Comment (ouchadam) [reply library/core-utils/src/main/java/im/vector/lib/core/utils/compat/MutableCollectionCompat.kt:25]: thanks for checking!

**Comment by bmarty** [comment]
Now included in #4971

**Pull request closed without merging**